planner/src/main/codegen/includes/show.ftl

```diff
@@ -46,21 +46,33 @@ SqlNode SqlShowColumns() :
     }
 }
 
-// DESCRIBE "table"
+// DESCRIBE TABLE "table"
```

Comment:
While I generally like to be as specific as I can, I am a bit worried because `DESCRIBE <table>`  is actually a well-known SQL command (see e.g. [here](https://www.geeksforgeeks.org/sql-describe-statement/#:~:text=So%20desc%20or%20describe%20command,time%20of%20Creation%20of%20table.)). How about allowong both `DESCRIBE TABLE <table>` and `DESCRIBE <table>` (and also `DESCRIBE MODEL <model>`)? You can use a or-clause (with a `|`) to distringuish the different cases.
(Feel free to ask me, if that was not clear enough).

Reply:
I may want your input here @nils-braun , Actually I tried with `DESCRIBE TABLE <table_name>` and `DESCRIBE <table_name>` , But I think  parser on seeing "DESCRIBE" it automatically redirects to `ShowColumnsPlugin ` so `DESCRIBE MODEL <model_name>` was parsed incorrectly that's y added TABLE keyword to distinguish between this two situation , please let me know if some other things I can try to rectify this situation .
Please find below screenshot for further information.
<img width="972" alt="describeTable" src="https://user-images.githubusercontent.com/25167333/119633332-42e55c00-be2f-11eb-80c6-46313d8e5e7a.png">
<img width="1007" alt="MODELkeyword-interpretedasTableName" src="https://user-images.githubusercontent.com/25167333/119633393-542e6880-be2f-11eb-8a51-430171e9d193.png">
<img width="1007" alt="ThisErrorisExpected" src="https://user-images.githubusercontent.com/25167333/119633486-6b6d5600-be2f-11eb-9aaf-dce710f1577c.png">

Reply:
Sure! Let me try to give you some more context. 
The `ftl` files are Java freemarker template files. They are turned into a parsing logic by a maven plugin, which is then integrated into the jar. There exist multiple ways how to specify the grammar you want to describe, and one is also the "alternative". It looks similar to this:
```freemarker
SqlNode Something() :
{
    ....
}
{
    <DESCRIBE>
    (
        tableName = CompoundTableIdentifier()
        {
            return ...
        }
    |
        <MODEL>
        modelIdentifier = ...
        {
            return ...
         }
    )
}
```
With that, we define two possibilities: `DESCRIBE "table"` and `DESCRIBE MODEL "model"`, which could also return different Java classes (and they should :-)).
 It would even be possible to include an optional `[ <TABLE> ]`, to also allow for a `DESCRIBE TABLE "table"`.

One thing to note when writing these alternatives (or branches), is that the parser will decide on the very first token, which route to take. This means, if you have two alternatives `A B C` and `A B D`, it can not decide after `A` already. One way to circumvent this, is to use a `LOOKAHEAD(2)` command, which will tell the parser to decide after two tokens.

So to summarize: you need to combine the model and table `DESCRIBE` statements into a single function, which will look similar to this:
```freemarker
SqlNode SqlDescribeTableOrModel() :
{
    final Span s;
    final SqlIdentifier schemaName;
    final SqlIdentifier tableName;
    final SqlModelIdentifier modelName;
}
{
    <DESCRIBE> { s = span(); }
    (
        LOOKAHEAD(2)
        modelName = ModelIdentifier()
        {
            return new SqlShowModelParams(s.end(this), modelName);
        }
    |
        [ <TABLE> ]
        tableName = CompoundTableIdentifier()
        {
            return new SqlShowColumns(s.end(this), tableName);
        }
    )
}
```
(please note that I did not test the code, so there might be some syntax errors)

Reply:
Please also not the order of the two branches: first check the `MODEL` alternative.

Reply:
wow, Thanks for the great explanation, Trying this now , will let you know shortly .